build.zig

```diff
@@ -459,17 +490,65 @@ pub fn build(b: *Builder) !void {
     spectests_step.dependOn(&run_spectests.step);
 }
 
-fn build_rust_project(b: *Builder, path: []const u8) *Builder.Step.Run {
-    return b.addSystemCommand(&.{
-        "cargo",
-        "+nightly",
-        "-C",
-        path,
-        "-Z",
-        "unstable-options",
-        "build",
-        "--release",
-    });
+fn build_rust_project(b: *Builder, path: []const u8, prover: ProverChoice) *Builder.Step.Run {
+    // Build only the selected prover crates
+    // Use optimized profiles for single-prover builds to reduce binary size
+    const cargo_build = switch (prover) {
+        .none => b.addSystemCommand(&.{
+            "cargo", "+nightly",  "-C", path,          "-Z", "unstable-options",
+            "build", "--release", "-p", "libp2p-glue",
+        }),
+        .risc0 => b.addSystemCommand(&.{
+            "cargo",      "+nightly",  "-C",            path, "-Z",          "unstable-options",
+            "build",      "--profile", "risc0-release", "-p", "libp2p-glue", "-p",
+            "risc0-glue",
+        }),
+        .openvm => b.addSystemCommand(&.{
+            "cargo",       "+nightly",  "-C",             path, "-Z",          "unstable-options",
+            "build",       "--profile", "openvm-release", "-p", "libp2p-glue", "-p",
+            "openvm-glue",
+        }),
+        .all => b.addSystemCommand(&.{
+            "cargo", "+nightly",  "-C",    path, "-Z", "unstable-options",
+            "build", "--release", "--all",
+        }),
+    };
+
+    // Weaken allocation symbols in built libraries to avoid duplicate symbol errors
+    // Using wildcards (-w) to match symbols across different Rust versions
+    const libs = switch (prover) {
+        .none => "rust/target/release/liblibp2p_glue.a",
+        .risc0 => "rust/target/risc0-release/librisc0_glue.a rust/target/risc0-release/liblibp2p_glue.a",
+        .openvm => "rust/target/openvm-release/libopenvm_glue.a rust/target/openvm-release/liblibp2p_glue.a",
+        .all => "rust/target/release/librisc0_glue.a rust/target/release/liblibp2p_glue.a rust/target/release/libopenvm_glue.a",
+    };
+
+    const weaken_cmd = b.fmt(
+        \\for lib in {s}; do \
+        \\  if [ -f "./$lib" ]; then \
+        \\    objcopy -w \
```

Comment:
looks like obcopy doesn't work on mac, at least not that installed version on this specific macos